src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -53,18 +53,32 @@
      */
     private int maxCitizens = Configurations.gameplay.maxCitizens;
 
+    /**
+     * The colony of the manager.
+     */
+    private final Colony colony;
+
+    /**
+     * Creates the Citizenmanager for a colony.
+     * @param colony the colony.
+     */
+    public CitizenManager(final Colony colony)
+    {
+        this.colony = colony;
+    }
+
     @Override
-    public void readFromNBT(@NotNull final NBTTagCompound compound, @NotNull final Colony colony)
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
     {
         maxCitizens = compound.getInteger(TAG_MAX_CITIZENS);
 
         //  Citizens before Buildings, because Buildings track the Citizens
         citizens.putAll(NBTUtils.streamCompound(compound.getTagList(TAG_CITIZENS, Constants.NBT.TAG_COMPOUND))
-                .map(citizenCompound -> deserializeCitizen(citizenCompound, colony))
+                .map(this::deserializeCitizen)
                 .collect(Collectors.toMap(CitizenData::getId, citizen -> citizen)));
```

Comment:
`citizen -> citizen` can be `Function::identity` https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--

Reply:
It seems it has to be Function.identity